src/ashley/api/views.py

```diff
@@ -0,0 +1,124 @@
+"""Views of the ashley api django application."""
+import logging
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Group
+from django.core.exceptions import PermissionDenied
+from machina.apps.forum_permission.viewmixins import (
+    PermissionRequiredMixin as BasePermissionRequiredMixin,
+)
+from machina.core.db.models import get_model
+from machina.core.loading import get_class
+from rest_framework import mixins, status, viewsets
+from rest_framework.response import Response
+
+from ashley.context_mixins import get_current_lti_session
+from ashley.defaults import (
+    _FORUM_ROLE_INSTRUCTOR,
+    _FORUM_ROLE_MODERATOR,
+    _FORUM_ROLE_STUDENT,
+)
+from ashley.permissions import ManageModeratorPermission
+
+from .serializers import UserSerializer
+
+User = get_user_model()
+LTIContext = get_model("ashley", "LTIContext")
+Forum = get_model("forum", "Forum")
+PermissionRequiredMixin: BasePermissionRequiredMixin = get_class(
+    "forum_permission.viewmixins", "PermissionRequiredMixin"
+)
+logger = logging.getLogger(__name__)
+
+
+class UserApiView(
+    mixins.ListModelMixin, mixins.UpdateModelMixin, viewsets.GenericViewSet
+):  # pylint: disable=too-many-ancestors
+    """API requests to list moderators for current LTIContext"""
+
+    # Add permission to check if user has the right permission and can manage moderators
+    permission_classes = [ManageModeratorPermission]
+    serializer_class = UserSerializer
+    ordering = "public_username"
+
+    def get_queryset(self):
+        context = get_current_lti_session(self.request)
+        role = self.request.GET.get("role", False)
+
+        if role is not False:
```

Comment:
This `if` is useless and can be removed. Or you can do the opposite and then remove the block line 83:

```suggestion
        if role is False:
            # If no role is specified, filter all the users for the current LTIContext
            return (
                User.objects.filter(
                    groups__name__startswith=context.get_group_role_name("")
                )
                .order_by(self.ordering)
                .distinct()
            )
```        
```

Reply:
lint made me do it this way.
It wants a return statement and it bothers me to do a return statement at the end for a default case which means role is False letting the code testing 3 times role for nothing.

Reply:
![](https://media.giphy.com/media/3o7527pa7qs9kCG78A/giphy.gif)

Reply:
so quiet

Reply:
And something like this ? 

```
def get_queryset(self):
        context = get_current_lti_session(self.request)
        role = self.request.GET.get("role", False)

        # Filter all the users that are moderators or students."""
        if role == _FORUM_ROLE_STUDENT:
            # Gets all the students for the current LTIContext.
            return (
                User.objects.filter(
                    is_active=True,
                    groups__name=context.get_group_role_name(_FORUM_ROLE_STUDENT),
                )
                .exclude(
                    groups__name=context.get_group_role_name(_FORUM_ROLE_MODERATOR)
                )
                .order_by(self.ordering)
            )
        if role == _FORUM_ROLE_MODERATOR:
            # Gets all the moderators for the current LTIContext.Moderators have
            # the primitive role student."""
            return (
                User.objects.filter(
                    is_active=True,
                    groups__name=context.get_group_role_name(_FORUM_ROLE_MODERATOR),
                )
                .filter(groups__name=context.get_group_role_name(_FORUM_ROLE_STUDENT))
                .order_by(self.ordering)
            )
        if role == _FORUM_ROLE_INSTRUCTOR:
            # Gets all the instructor for the current LTIContext.
            return User.objects.filter(
                is_active=True,
                groups__name=context.get_group_role_name(_FORUM_ROLE_INSTRUCTOR),
            ).order_by(self.ordering)

        return (
            User.objects.filter(
                groups__name__startswith=context.get_group_role_name("")
            )
            .order_by(self.ordering)
            .distinct()
        )
```

Reply:
it's what I was telling you, our code will test three times the value of role when it's not even defined it it is the case, that was the point that was bothering me

Reply:
I think it's clearer to limit the number of `if` indentation. the `if role is not False` is really useless I think.